File: core/picture/lib/src/main/java/org/pcsoft/app/jmc/picture/ui/panes/PictureProjectPane.java
package org.pcsoft.app.jmc.picture.ui.panes;

import javafx.beans.property.ObjectProperty;
import javafx.scene.layout.BorderPane;
import org.pcsoft.framework.jfex.mvvm.Fxml;

import java.io.File;

public class PictureProjectPane extends BorderPane {
    private final PictureProjectPaneView viewController;
    private final PictureProjectPaneViewModel viewModel;

    public PictureProjectPane() {
        final var tuple = Fxml.from(PictureProjectPaneView.class).withRoot(this).load();
        viewController = tuple.getViewController();
        viewModel = tuple.getViewModel();
    }

    public PictureProjectPane(File file) {
        this();
        viewModel.setFile(file);
    }

    public File getFile() {
        return viewModel.getFile();
    }

    public ObjectProperty<File> fileProperty() {
        return viewModel.fileProperty();
    }

    public void setFile(File file) {
        viewModel.setFile(file);
    }
}

File: core/picture/lib/src/main/kotlin/org/pcsoft/app/jmc/picture/ui/panes/PictureProjectPane.kt
package org.pcsoft.app.jmc.picture.ui.panes

import javafx.scene.layout.BorderPane
import org.pcsoft.framework.jfex.mvvm.Fxml
import java.io.File

class PictureProjectPane(file: File) : BorderPane() {
    private val viewController: PictureProjectPaneView
    private val viewModel: PictureProjectPaneViewModel

    init {
        val tuple = Fxml.from(PictureProjectPaneView::class.java).withRoot(this).load()
        viewController = tuple.viewController
        viewModel = tuple.viewModel

        viewModel.file.set(file)
    }

    val file get() = viewModel.file
}